AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/backgroundblur/BackgroundBlurVideoFrameProcessor.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  BackgroundBlurVideoFrameProcessor.swift
+//  AmazonChimeSDK
+//
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+
+import CoreImage
+import UIKit
+
+/// `BackgroundBlurVideoFrameProcessor` is a processor which receives video frames via `VideoSource` and
+/// then applies Gaussian blur to each video frame and renders the foreground on top of the blurred image.
+/// Gaussian blur is applied using built in `CIGaussianBlur` CIFilter.
+@objcMembers public class BackgroundBlurVideoFrameProcessor: VideoSource, VideoSink {
+    public var videoContentHint = VideoContentHint.motion
+
+    /// Context used for processing and rendering the final output image.
+    private let context = CIContext(options: [.cacheIntermediates: false])
+
+    /// Set of `VideoSource` sinks.
+    private let sinks = ConcurrentMutableSet()
+
+    /// Blur strength value to adjust blur intensity.
+    private var blurStrength: Int
+
+    /// Background filter processor.
+    private let backgroundFilterProcessor: BackgroundFilterProcessor
+
+    /// Public constructor to initialize the processor.
+    ///
+    /// - Parameters:
+    ///   - backgroundBlurConfiguration: `BackgroundBlurConfiguration` class.
+    public init(backgroundBlurConfiguration: BackgroundBlurConfiguration) {
+        self.blurStrength = backgroundBlurConfiguration.blurStrength.rawValue
+        self.backgroundFilterProcessor = backgroundBlurConfiguration.backgroundFilterProcessor
+    }
+
+    /// Receive a video frame from some upstream source. The foreground is segmented and then masked
+    /// on top of the blurred background.
+    ///
+    /// - Parameters:
+    ///   - frame: New video frame to consume.
+    public func onVideoFrameReceived(frame: VideoFrame) {
+        var processedFrame: VideoFrame = frame
+
+        // Update sinks at the end.
+        defer {
+            updateSinks(frame: processedFrame)
+        }
+
+        // Create `VideoFramePixelBuffer` of the input frame.
+        guard let pixelBuffer = frame.buffer as? VideoFramePixelBuffer else {
+            backgroundFilterProcessor.logger.debug(debugFunction: {
```

Comment:
any reason why using `backgroundFilterProcessor.logger` instead of having its own logger in this class?

Reply:
It was suggested that only one logger is passed as a dependency down to `backgroundFilterProcessor` and the `BackgroundBlurVideoFrameProcessor` and `BackgroundReplacementVideoFrameProcessor` can just reference it. It also gives the builder more control on the logger.

Reply:
Even if the logger is passed to this class, it will be the same logger as the `backgroundFilterProcessor.logger`. I was suggesting to reference to its parent's logger when you were using inheritance(iirc `backgroundFilterProcessor` is used to be parent class of `BackgroundBlurVideoFrameProcessor`), since you changed the pattern to composition, each processor should maintain their own `logger` reference.

Reply:
added a reference to the logger and used that instead of `backgroundFilterProcessor.logger`